src/Parser.js

```diff
@@ -598,49 +570,20 @@ export default class Parser {
     parseStringGroup(
         modeName: ArgType,  // Used to describe the mode in error messages.
         optional: boolean,
-        raw?: boolean,
     ): ?Token {
-        const groupBegin = optional ? "[" : "{";
-        const groupEnd = optional ? "]" : "}";
-        const beginToken = this.fetch();
-        if (beginToken.text !== groupBegin) {
-            if (optional) {
-                return null;
-            } else if (raw && beginToken.text !== "EOF" &&
-                    /[^{}[\]]/.test(beginToken.text)) {
-                this.consume();
-                return beginToken;
-            }
+        const argToken = this.gullet.scanArgument(optional);
+        if (argToken == null) {
+            return null;
         }
-        const outerMode = this.mode;
-        this.mode = "text";
-        this.expect(groupBegin);
         let str = "";
-        const firstToken = this.fetch();
-        let nested = 0; // allow nested braces in raw string group
-        let lastToken = firstToken;
         let nextToken;
-        while ((nextToken = this.fetch()).text !== groupEnd ||
-               (raw && nested > 0)) {
-            switch (nextToken.text) {
-                case "EOF":
-                    throw new ParseError(
-                        "Unexpected end of input in " + modeName,
-                        firstToken.range(lastToken, str));
-                case groupBegin:
-                    nested++;
-                    break;
-                case groupEnd:
-                    nested--;
-                    break;
-            }
-            lastToken = nextToken;
-            str += lastToken.text;
+        while ((nextToken = this.fetch()).text !== "EOF") {
+            str += nextToken.text;
             this.consume();
         }
-        this.expect(groupEnd);
-        this.mode = outerMode;
-        return firstToken.range(lastToken, str);
+        this.consume(); // consume the end of the argument
+        argToken.text = str;
+        return argToken;
```

Comment:
If we replace the `.text` won't the `argToken.loc` be incorrect?

Reply:
@kevinbarabash Normally, they should match since the end of the argument is denoted using `EOF`. Otherwise, it's an error and will be caught by the `EOF` mismatch.

Reply:
@kevinbarabash Their lengths may not match, due to whitespace normalization, etc.